### PR TITLE
Android - UI - Detail Screen: Scaffold initial screen with elements#88

### PR DESCRIPTION
**Pull request by lsireneva**
Issue https://github.com/WomenWhoCode/WWCodeMobile/issues/88

Added Compose UI for Detail Screen:  
   -  added Date, Start time, and End time text
   -  added textbox for activity description
   -  added button that displays date
   - added button that displays start time
   - added button that displays end time
   - added delete button
   - added cancel button
   - added done button

Changes in files:
 - TaskDetailScreen.kt (added compose UI for Detail Screen)
 - Color.kt (added green color for buttons)
 - added 2 new drawables for image buttons (delete and cancel)
 - strings.xml (added string resources)
 - -dimens.xml (added dimensions)

Test:
At this moment only in design mode in Android studio, screenshot is attached to PR

detail_screen resized

### CODE REVIEW

File: coding-projects/android/TaskTracker/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,3 +1,10 @@
 <resources>
     <string name="app_name">Task Tracker</string>
+    <string name="date_label">DATE</string>
+    <string name="start_time_label">START TIME</string>
+    <string name="end_time_label">END TIME</string>
+    <string name="done">DONE</string>
```
Comment (gaurigadkari) [inline coding-projects/android/TaskTracker/app/src/main/res/values/strings.xml:6]: It is generally a good practice to store strings in their original form (non-capitalized) in the strings.xml file and then capitalize or format them as needed
getString(R.string. date_label).toUpperCase();

Comment (lsireneva) [reply coding-projects/android/TaskTracker/app/src/main/res/values/strings.xml]: @gaurigadkari I have updated strings.xml and TaskDetailScreen.kt

**Comment by lsireneva** [comment]
@renweview Could you please review?

File: coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt [hunk 2]
```diff
@@ -1,8 +1,168 @@
...
+
+                Image(
+                    painter = painterResource(id = R.drawable.baseline_cancel_24),
+                    contentDescription = stringResource(id = R.string.cancel)
+                )
+            }
+
+            LabelButtonRow(
+                label = stringResource(id = R.string.date_label).uppercase(),
+                buttonInfo = "JAN 29 2024"
```
Comment (gaurigadkari) [inline coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt:84]: can we remove these hardcoded strings

File: coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt [hunk 3]
```diff
@@ -1,8 +1,168 @@
...
+                    containerColor = Color.White,
+                    focusedIndicatorColor = Color.Transparent,
+                    unfocusedIndicatorColor = Color.Transparent,
+                    disabledIndicatorColor = Color.Transparent
+                ),
+            )
+
+            LabelButtonRow(
+                label = stringResource(id = R.string.start_time_label).uppercase(),
+                buttonInfo = "08:22:10"
```
Comment (gaurigadkari) [inline coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt:112]: can we remove these hardcoded strings

File: coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt [hunk 4]
```diff
@@ -1,8 +1,168 @@
...
+            ) {}
+            LabelButtonRow(
+                label = stringResource(id = R.string.end_time_label).uppercase(),
+                buttonInfo = "09:12:01"
```
Comment (gaurigadkari) [inline coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt:116]: can we remove these hardcoded strings

File: coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt [hunk 5]
```diff
@@ -1,8 +1,168 @@
...
\ No newline at end of file
+    @OptIn(ExperimentalMaterial3Api::class)
+    @Composable
+    fun TaskDetailScreen() {
+        OutlinedCard(
+            colors = CardDefaults.cardColors(
+                containerColor = Color.White,
+            ),
+            border = BorderStroke(1.dp, Color.Black),
+            modifier = Modifier
```
Comment (gaurigadkari) [inline coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt:57]: lets also add all these dimensions to the dimens file

**Comment by lsireneva** [comment]
@gaurigadkari Hi Gauri, I have made the suggested changes, please check it out when you get a chance.

**Comment by vuinguyen** [comment]
@gaurigadkari @renweview I'm assisting with the PR review as needed. I get this from the Android Studio Compose preview window. Does this help?
Screenshot 2024-02-06 at 11 35 25 AM

File: coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt [hunk 6]
```diff
@@ -1,8 +1,172 @@
...
+            val dateInfo = "JAN 29 2024" //these values should be changed, it needs to be get from ViewModel
+            val startTimeInfo = "08:22:10"
+            val endTimeInfo = "09:12:01"
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(dimensionResource(R.dimen.detail_content_padding_16)),
+                horizontalArrangement = Arrangement.End
+            ) {
+                Image(
```
Comment (baghaii) [inline coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt:74]: Can these use the Icon composable and be wrapped in IconButton?

File: coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt [hunk 7]
```diff
@@ -1,8 +1,190 @@
...
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(dimensionResource(R.dimen.detail_content_padding_16)),
+                horizontalArrangement = Arrangement.End
+            ) {
+                IconButton(onClick = { /*TODO*/ }) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.outline_delete_24),
+                        contentDescription = stringResource(id = R.string.delete),
+                        modifier = Modifier.padding(horizontal = dimensionResource(R.dimen.detail_content_padding_8))
```
Comment (baghaii) [inline coding-projects/android/TaskTracker/app/src/main/java/com/example/tasktracker/ui/TaskDetail/TaskDetailScreen.kt:79]: The padding on this should probably be moved out of the icon into a Spacer between the two buttons. I am sorry that I did not catch this one earlier. I think that if it is in the button, this button will look bigger than the other one when you press it.

Overall, I think you have done a great job on this PR, and I hope that it gets merged soon.

**Review by baghaii — APPROVED** [review]

**Review by gaurigadkari — APPROVED** [review]

**Pull request merged**